aptos-move/block-executor/src/executor_utilities.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{errors::*, task::ExecutorTask, view::LatestView};
+use aptos_aggregator::types::{code_invariant_error, PanicError, PanicOr};
+use aptos_logger::error;
+use aptos_mvhashmap::types::ValueWithLayout;
+use aptos_types::{
+    contract_event::TransactionEvent, executable::Executable, state_store::TStateView,
+    transaction::BlockExecutableTransaction as Transaction, write_set::TransactionWrite,
+};
+use bytes::Bytes;
+use move_core_types::value::MoveTypeLayout;
+use rand::{thread_rng, Rng};
+use std::{
+    collections::{BTreeMap, HashSet},
+    sync::Arc,
+};
+
+macro_rules! groups_to_finalize {
```

Comment:
Why macro and not a function? I might have missed a use case where you pass additional arguments (`*`) though. Function can be easier to read?

Reply:
not just txn_idx argument, also argument type (just responded at the other place as well). I'd also prefer a function, maybe we can get there with some refactors (but would need some traits, etc)